### PR TITLE
company endpoints for b2b

### PR DESCRIPTION
**Pull request by AnastasiiaYefremkinaa**
Added TCs for the following endpoints:
- companies;
- company-business-unit-addresses;
- company-business-units;
- company-roles;
- company-user-access-tokens;
- company-users.

### CODE REVIEW

File: tests/api/b2b/glue/company_endpoints/company_users/positive.robot [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    ...    AND    I set Headers:    Content-Type=${default_header_content_type}    Authorization=${token}  
+    When I send a GET request:    /company-users
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Response header parameter should be:    Content-Type    ${default_header_content_type}
+    And Response should contain the array of a certain size:    [data]  12
+    And Each array element of array in response should contain property with value:    [data]    type   company-users   
+    And Each array element of array in response should contain property:    [data]    id
+    And Each array element of array in response should contain property:    [data]    attributes
+    And Response should contain the array of a certain size:    [data][0][attributes]  2
```
Comment (OKrivtsova) [inline tests/api/b2b/glue/company_endpoints/company_users/positive.robot:17]: why don't we check those 2 attribues?

File: tests/api/b2b/glue/company_endpoints/companies/positive.robot [hunk 2]
```diff
@@ -0,0 +1,33 @@
+*** Settings ***
```
[the remark this answers is not in the document]
Comment (AnastasiiaYefremkinaa) [reply tests/api/b2b/glue/company_endpoints/companies/positive.robot:1]: created a ticket: https://spryker.atlassian.net/browse/CC-16003

File: tests/api/b2b/glue/company_endpoints/companies/positive.robot [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...    AND    I set Headers:    Content-Type=${default_header_content_type}    Authorization=${token}  
+    When I send a GET request:    /companies/${company_id}
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Response header parameter should be:    Content-Type    ${default_header_content_type}
+    And Response body parameter should be:    [data][type]    companies
+    And Response body parameter should be:    [data][id]    ${company_id}
+    And Response body parameter should be:    [data][attributes][name]    ${company_name}
+    And Response body parameter should be:    [data][attributes][status]    approved
```
[the remark this answers is not in the document]
Comment (AnastasiiaYefremkinaa) [reply tests/api/b2b/glue/company_endpoints/companies/positive.robot:16]: added

File: tests/api/b2b/glue/company_endpoints/companies/negative.robot [hunk 4]
```diff
@@ -0,0 +1,30 @@
+*** Settings ***
```
[the remark this answers is not in the document]
Comment (AnastasiiaYefremkinaa) [reply tests/api/b2b/glue/company_endpoints/companies/negative.robot:1]: added

File: tests/api/b2b/glue/company_endpoints/companies/positive.robot [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    And Response should contain the array of a certain size:    [data][attributes]    3
+
+
+Request_company_by_mine
```
[the remark this answers is not in the document]
Comment (AnastasiiaYefremkinaa) [reply tests/api/b2b/glue/company_endpoints/companies/positive.robot:20]: added Request_companies_users_if_user_has_4_companies test to /company-users endpoint

File: tests/api/b2b/glue/company_endpoints/companies/positive.robot [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    ...    AND    I set Headers:    Content-Type=${default_header_content_type}    Authorization=${token}  
+    When I send a GET request:    /companies/mine
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Response header parameter should be:    Content-Type    ${default_header_content_type}
+    And Response body parameter should be:    [data][0][type]    companies
+    And Response body parameter should be:    [data][0][id]    ${company_id}
+    And Response body parameter should be:    [data][0][attributes][name]    ${company_name}
+    And Response body parameter should be:    [data][0][attributes][status]    approved
+    And Response should contain the array of a certain size:    [data][0][attributes]    3
```
[the remark this answers is not in the document]
Comment (AnastasiiaYefremkinaa) [reply tests/api/b2b/glue/company_endpoints/companies/positive.robot:31]: added

File: tests/api/b2b/glue/company_endpoints/company_business_unit/negative.robot [hunk 7]
```diff
@@ -0,0 +1,30 @@
+*** Settings ***
```
[the remark this answers is not in the document]
Comment (AnastasiiaYefremkinaa) [reply tests/api/b2b/glue/company_endpoints/company_business_unit/negative.robot:1]: added

File: tests/api/b2b/glue/company_endpoints/company_business_unit/positive.robot [hunk 8]
```diff
@@ -0,0 +1,17 @@
+*** Settings ***
```
[the remark this answers is not in the document]
Comment (AnastasiiaYefremkinaa) [reply tests/api/b2b/glue/company_endpoints/company_business_unit/positive.robot:1]: also added to ticket https://spryker.atlassian.net/browse/CC-16003

File: tests/api/b2b/glue/company_endpoints/company_business_unit/positive.robot [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+Suite Setup       SuiteSetup
+Resource    ../../../../../../resources/common/common_api.robot
+
+*** Test Cases ***
+Request_business_unit_by_id
```
[the remark this answers is not in the document]
Comment (AnastasiiaYefremkinaa) [reply tests/api/b2b/glue/company_endpoints/company_business_unit/positive.robot:6]: added

File: tests/api/b2b/glue/company_endpoints/company_business_unit_addresses/positive.robot [hunk 10]
```diff
@@ -0,0 +1,16 @@
+*** Settings ***
```
[the remark this answers is not in the document]
Comment (AnastasiiaYefremkinaa) [reply tests/api/b2b/glue/company_endpoints/company_business_unit_addresses/positive.robot:1]: also added to ticket https://spryker.atlassian.net/browse/CC-16003

File: tests/api/b2b/glue/company_endpoints/company_business_unit_addresses/positive.robot [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+Request_business_units_address_by_id
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...    AND    I set Headers:    Content-Type=${default_header_content_type}    Authorization=${token}  
+    When I send a GET request:    /company-business-unit-addresses/${busines_unit_address_id}
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Response header parameter should be:    Content-Type    ${default_header_content_type}
+    And Response body parameter should be:    [data][type]    company-business-unit-addresses
+    And Response body parameter should be:    [data][id]    ${busines_unit_address_id}
+    And Response should contain the array of a certain size:    [data][attributes]  8
```
[the remark this answers is not in the document]
Comment (AnastasiiaYefremkinaa) [reply tests/api/b2b/glue/company_endpoints/company_business_unit_addresses/positive.robot:15]: fixed

File: tests/api/b2b/glue/company_endpoints/company_roles/negative.robot [hunk 12]
```diff
@@ -0,0 +1,28 @@
+*** Settings ***
+Suite Setup       SuiteSetup
```
[the remark this answers is not in the document]
Comment (AnastasiiaYefremkinaa) [reply tests/api/b2b/glue/company_endpoints/company_roles/negative.robot:2]: added

File: tests/api/b2b/glue/company_endpoints/company_roles/positive.robot [hunk 13]
```diff
@@ -0,0 +1,17 @@
+*** Settings ***
+Suite Setup       SuiteSetup
+Resource    ../../../../../../resources/common/common_api.robot
+
+*** Test Cases ***
+Request_company_role_by_id
```
[the remark this answers is not in the document]
Comment (AnastasiiaYefremkinaa) [reply tests/api/b2b/glue/company_endpoints/company_roles/positive.robot:6]: added tests, ticket to update swagger: https://spryker.atlassian.net/browse/CC-16003

File: tests/api/b2b/glue/company_endpoints/company_user_access_tokens/negative.robot [hunk 14]
```diff
@@ -0,0 +1,39 @@
+*** Settings ***
+Suite Setup       SuiteSetup
```
[the remark this answers is not in the document]
Comment (AnastasiiaYefremkinaa) [reply tests/api/b2b/glue/company_endpoints/company_user_access_tokens/negative.robot:2]: added

File: tests/api/b2b/glue/company_endpoints/company_users/positive.robot (hunk 1 above)
Comment (AnastasiiaYefremkinaa) [reply tests/api/b2b/glue/company_endpoints/company_users/positive.robot:17]: added

File: tests/api/b2b/glue/company_endpoints/company_users/positive.robot [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Response header parameter should be:    Content-Type    ${default_header_content_type}   
+    And Response should contain the array of a certain size:    [data]  1
+    And Response body parameter should be:    [data][0][type]   company-users   
+    And Response body parameter should not be EMPTY:    [data][0][id]
+    And Response body parameter should be:    [data][0][attributes][isActive]    True
+    And Response body parameter should be:    [data][0][attributes][isDefault]    False
+
+Request_company_users_include_customers_and_roles_and_business_units
```
[the remark this answers is not in the document]
Comment (AnastasiiaYefremkinaa) [reply tests/api/b2b/glue/company_endpoints/company_users/positive.robot:43]: added tests

File: tests/api/b2b/glue/company_endpoints/company_business_unit_addresses/positive.robot (hunk 11 above)
[the remark this answers is not in the document]
Comment (OKrivtsova) [reply tests/api/b2b/glue/company_endpoints/company_business_unit_addresses/positive.robot:16]: additionally fixed hardcode for 'not empty' check

**Review by OKrivtsova — APPROVED** [review]

**Review by OKrivtsova — APPROVED** [review]

**Pull request merged**